staging_vespalib/src/vespa/vespalib/metrics/clock.h

```diff
@@ -1,12 +1,36 @@
 // Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 #pragma once
 
-#include "wallclock_time_supplier.h"
+#include <chrono>
+#include <memory>
 
-namespace vespalib {
-namespace metrics {
+namespace vespalib::metrics {
 
-using InternalTimeStamp = WallclockTimeSupplier::TimeStamp;
+using TimeStamp = std::chrono::duration<double, std::ratio<1,1>>;
 
-} // namespace metrics
-} // namespace vespalib
+/**
+ * Simple interface abstracting both timing and time measurement for
+ * threads wanting to do stuff at regular intervals and also knowing
+ * at what time stuff was done. The 'next' function blocks until the
+ * next tick is due and returns the current number of seconds since
+ * epoch. The parameter passed to the 'next' function should be its
+ * previous return value, except the first time it is called, then 0
+ * should be used. A convenience function called 'first' is added for
+ * this purpose.
+ **/
+struct Tick {
+    using UP = std::unique_ptr<Tick>;
+    virtual TimeStamp next(TimeStamp prev) = 0;
+    TimeStamp first() { return next(TimeStamp(0.0)); }
+    virtual void kill() = 0;
```

Comment:
consider adding an 'alive' function to this interface and let the Tick control thread stopping. also use this to avoid calling the internal 'tick' function after the Tick has been killed.